### PR TITLE
Move to GitHub actions

### PR DESCRIPTION
**Pull request by rhruiz**
Also bumps some dependencies and makes credo execution conditional to the Elixir version.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -42,6 +46,9 @@ defmodule Cassette.Mixfile do
     ]
   end
 
+  @elixir_version Version.parse!(System.version())
```
Comment (VictorSNA) [inline mix.exs:49]: Duped line here! It's already at line 4!

File: mix.exs [hunk 2]
```diff
@@ -58,9 +65,15 @@ defmodule Cassette.Mixfile do
       {:ex_doc, "~> 0.11", only: :dev},
       {:earmark, "~> 1.0", only: :dev},
       {:bypass, "~> 1.0", only: [:dev, :test]},
-      {:credo, "~> 1.0", only: [:dev, :test], runtime: false},
       {:fake_cas, "~> 1.1", only: [:dev, :test]},
       {:excoveralls, "~> 0.7", only: :test}
     ]
+    |> (fn deps ->
+          if Version.compare(@elixir_version, @min_version_for_credo) in [:gt, :eq] do
```
Comment (VictorSNA) [inline mix.exs:72]: WDYT adding to do telling when it can be removed?

**Review by VictorSNA — COMMENTED** [review]
Nice job!

**Review by rogesson — APPROVED** [review]
Nice!

File: mix.exs (hunk 1 above)
Comment (rhruiz) [reply mix.exs]: fixed :)

File: mix.exs (hunk 2 above)
Comment (rhruiz) [reply mix.exs:72]: Do you think it is clear now?

**Review by TecnoSigma — APPROVED** [review]
Very good!

**Review by VictorSNA — APPROVED** [review]

**Pull request merged**